webpack.config.js

```diff
@@ -0,0 +1,245 @@
+let fs = require('fs');
+let path = require('path');
+let webpack = require('webpack');
+// let ExtractTextPlugin = require('extract-text-webpack-plugin');
+let CleanWebpackPlugin = require('clean-webpack-plugin');
+let ProgressBarPlugin = require('progress-bar-webpack-plugin');
+let MiniCssExtractPlugin = require('mini-css-extract-plugin');
+let HtmlWebpackPlugin = require('html-webpack-plugin');
+let PreloadWebpackPlugin = require('preload-webpack-plugin');
+let ReplacePlugin = require('webpack-plugin-replace');
+let CopyPlugin = require('copy-webpack-plugin');
+let WatchTimestampsPlugin = require('./config/watch-timestamps-plugin');
+let BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+
+let oldWrite = process.stderr.write;
+process.stderr.write = chunk => {
+	if (String(chunk).indexOf('DeprecationWarning:')!==-1) return;
+	return oldWrite.call(process.stderr, chunk);
+};
+
+function parseJson(text, fallback) {
+	try {
+		return JSON.parse(text);
+	}
+	catch (e) { }
+	return fallback || {};
+}
+
+function readFile(filename) {
+	try {
+		return fs.readFileSync(filename);
+	}
+	catch (e) {}
+}
+
+module.exports = function(_, env) {
+	let isProd = env.mode === 'production';
+	let nodeModules = path.join(__dirname, 'node_modules');
+	let componentStyleDirs = [
+		path.join(__dirname, 'src/components'),
+		path.join(__dirname, 'src/routes')
+	];
+
+	let babelRc = parseJson(readFile('.babelrc'));
+	babelRc.babelrc = false;
+	if (isProd) babelRc.presets[0][1].modules = false;
+
+	let manifest = parseJson(readFile('./src/manifest.json'));
+
+	return {
+		mode: env.mode || 'development',
+		entry: path.join(__dirname, 'config/client-boot.js'),
+		// entry: './src/index',
+		output: {
+			filename: isProd ? '[name].[chunkhash:5].js' : '[name].js',
+			chunkFilename: '[name].chunk.[chunkhash:5].js',
+			path: path.join(__dirname, 'build'),
+			publicPath: '/'
+		},
+		resolve: {
+			extensions: ['.ts', '.tsx', '.js', '.jsx', '.scss', '.css'],
+			alias: {
+				'app-entry-point': path.join(__dirname, 'src/index'),
+				style: path.join(__dirname, 'src/style'),
+				preact$: path.join(__dirname, 'node_modules/preact/dist/preact.js')
```

Comment:
This just uses the compressed version of Preact instead of the ESM version, since the compressed version is slightly smaller when bundled.  We could drop it, I doubt it saves more than a few bytes.

Reply:
We reviewed this file first. Later on we found the aliases really confusing honestly. It took us a while to realize that “app-entry-point” was not a module but an alias, for example. I think we should use them sparingly.

Reply:
Yup - the entrypoint one is important because it makes generic HMR possible, but the style one is likely unnecessary, and the Preact one can  be removed.

Reply:
Maybe that’s better for tomorrow: How much do you value/enjoy HMR during development? I am very skeptical (I have _never_ used it so all this is just me having feelings) and it seems brittle.